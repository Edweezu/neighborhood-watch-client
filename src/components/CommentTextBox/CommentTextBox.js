import React from 'react'
import MainContext from '../../contexts/MainContext';
import config from '../../config'
import TokenService from '../../services/token-service'

class CommentTextBox extends React.Component {
    static contextType = MainContext

    handleSubmit = (e) => {
        e.preventDefault()
        const { postId } = this.props
        //forgot this
        const text = e.target['CommentTextBox__text']

        const newComment = {
            text: text.value,
            post_id: postId
        }
       
        return fetch(`${config.API_ENDPOINT}/comments`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`
            },
            body: JSON.stringify(newComment)
        })
        .then(res => {
            if (!res.ok) {
                return res.json().then(e => Promise.reject(e))
            }
            return res.json()
        })
        .then(responseJson => {
            //forgot how to erase the value
            text.value = ''
            this.context.addComment(responseJson)
        })
        .catch(err => {
            console.error(err)
        })


    }

    render () {
        return (
            <form  onSubmit={this.handleSubmit}className='CommentTextBox'>
                <div>
                    <textarea
                        required 
                        aria-label='Type a comment...'
                        name='CommentTextBox__text'
                        id='CommentTextBox__text'
                        cols='30'
                        rows='3'
                        placeholder='Type a comment...'
                    >
                    </textarea>
                    <div>
                        <button type='submit'>Post Comment</button>
                    </div>
                </div>
            </form>
        )
    }
}

export default CommentTextBox