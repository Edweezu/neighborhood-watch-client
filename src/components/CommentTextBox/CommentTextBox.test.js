import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import CommentTextBox from './CommentTextBox'


describe('CommentTextBox component', () => {
    const props = {
        postId: 1
    }

    it('renders the given component', () => {
        const wrapper = shallow(<CommentTextBox />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders the component's given props`, () => {
        const wrapper = shallow(<CommentTextBox {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})