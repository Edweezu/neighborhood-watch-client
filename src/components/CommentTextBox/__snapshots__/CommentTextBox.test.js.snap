// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentTextBox component renders the component's given props 1`] = `
<form
  className="CommentTextBox"
  onSubmit={[Function]}
>
  <div>
    <textarea
      aria-label="Type a comment..."
      cols="30"
      id="CommentTextBox__text"
      name="CommentTextBox__text"
      placeholder="Type a comment..."
      required={true}
      rows="3"
    />
    <div>
      <button
        type="submit"
      >
        Post Comment
      </button>
    </div>
  </div>
</form>
`;

exports[`CommentTextBox component renders the given component 1`] = `
<form
  className="CommentTextBox"
  onSubmit={[Function]}
>
  <div>
    <textarea
      aria-label="Type a comment..."
      cols="30"
      id="CommentTextBox__text"
      name="CommentTextBox__text"
      placeholder="Type a comment..."
      required={true}
      rows="3"
    />
    <div>
      <button
        type="submit"
      >
        Post Comment
      </button>
    </div>
  </div>
</form>
`;
